examples/matrixMultiply.cpp

```diff
@@ -0,0 +1,247 @@
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+#include <algorithm>
+#include <initializer_list>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/index/RangeSegment.hpp"
+#include "RAJA/util/defines.hpp"
+
+#include "memoryManager.hpp"
+
+const int DIM = 2;
+void checkSolution(double *C, int in_N);
+void checkSolution(RAJA::View<double,RAJA::Layout<DIM> > Cview, int in_N);
+
+/*
+  Example 2: Multiplying Two Matrices
+
+  ----[Details]--------------------
+  Multiplies two N x N matrices
+
+  -----[RAJA Concepts]-------------
+  1. Nesting of forall loops (Not currently supported in CUDA)
+  2. ForallN variant of the RAJA loop
+  3. RAJA View
+*/
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  printf("Example 2: Multiplying Two N x N Matrices \n \n");
+  const int N = 1000;
+  const int NN = N * N;
+
+  double *A = memoryManager::allocate<double>(NN);
+  double *B = memoryManager::allocate<double>(NN);
+  double *C = memoryManager::allocate<double>(NN);
+
+  for(int i=0; i<NN; ++i){
+    A[i] = 1.0; 
+    B[i] = 1.0; 
+  }
+
+  
+  printf("Standard C++ Loop \n");
+  for (int r = 0; r < N; ++r) {
+    for (int c = 0; c < N; ++c) {
+
+      int cId = c + r * N;
+      double dot = 0.0;
+
+      for (int k = 0; k < N; ++k) {
+
+        int aId = k + r * N;
+        int bId = c + k * N;
+        dot += A[aId] * B[bId];
+      }
+
+      C[cId] = dot;
+    }
+  }
+  checkSolution(C, N);
+
+  /*
+    RAJA::View - Introduces Multidimensional arrays
+  */  
+
+  RAJA::View<double, RAJA::Layout<DIM> > Aview(A, N, N);
+  RAJA::View<double, RAJA::Layout<DIM> > Bview(B, N, N);
+  RAJA::View<double, RAJA::Layout<DIM> > Cview(C, N, N);
+
+
+  printf("RAJA: Sequential Policy - Single forall \n");  
+  RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(0, N), [=](int row) {
+      for(int col = 0; col < N; ++col){
+        
+        double dot = 0.0;
+        for (int k = 0; k < N; ++k) {
+          dot += Aview(row,k)*Bview(k,col);
+        }
+        
+        Cview(row,col) = dot; 
+      }
+
+    });
+  checkSolution(Cview, N);
+
+
+  printf("RAJA: Sequential Policy - Nested forall \n");
+  /*
+    Forall loops may be nested under sequential and omp policies
+  */
+  RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(0, N), [=](int row) {
+    RAJA::forall<RAJA::seq_exec>(RAJA::RangeSegment(0, N), [=](int col) {
+
+      double dot = 0.0;
+      for (int k = 0; k < N; ++k) {
+        dot += Aview(row,k)*Bview(k,col);
+      }
+      
+      Cview(row,col) = dot;
+    });
+  });
+  checkSolution(Cview, N);
+
+  printf("RAJA: Sequential Policy RAJA - forallN \n");
+  /*
+    Nested forall loops may be collapsed into a single forallN loop
+  */
+  RAJA::forallN<RAJA::NestedPolicy<RAJA::ExecList<RAJA::seq_exec,
+                                                  RAJA::seq_exec>>>(
+  RAJA::RangeSegment(0,N), RAJA::RangeSegment(0,N), [=](int row, int col) {
```

Comment:
you've isolated the RangeSegment declaration to be before the call to forall in prior examples. Maybe the same should be done here.